src/android/camera_hal_config.cpp

```diff
@@ -357,15 +357,8 @@ CameraHalConfig::CameraHalConfig()
  */
 int CameraHalConfig::parseConfigurationFile()
 {
-	std::filesystem::path filePath = LIBCAMERA_SYSCONF_DIR;
-	filePath /= "camera_hal.yaml";
-	if (!std::filesystem::is_regular_file(filePath)) {
```

Comment:
Also you've dropped a check, you're no longer validating that the file is there before opening it. I guess the difference is subtle as the fopen will likely fail anyway, but it will be a different error message reported. Looks like previously that only reports a Debug level, so maybe this is actually better...!?

Reply:
So should I fstat() it and check is it a regular file or  better to keep as is?